pkg/bosh/manifest/manifest.go

```diff
@@ -179,7 +179,7 @@ func (m *Manifest) loadDNS() {
 			var err error
 			dns, err := NewBoshDomainNameService(addon, m.Name)
 			if err != nil {
-				ctxlog.Errorf(context.TODO(), "Error loading bosh dns configuration: %s", err.Error())
+				ctxlog.Errorf(context.Background(), "Error loading bosh dns configuration: %v", err)
```

Comment:
Hmm are we on new enough golang to use `%w` yet? :D

(Is `Background` the correct context? Should it not be passed in somewhere?)

Reply:
Well, context.TODO is definitely not the correct context. There are lots of places that LoadYAML is called that doesn't have a context. I'd assume that it's ok to use it here?

Reply:
It's probably okay, yes, but there might have been better ones (that would have needed to be propagated). As a default, it's okay, but not really not _that_ much better than `TODO`.  At least `TODO` makes it clear there's more thinking required.

Reply:
You are right. It makes more sense to actually leave the TODO. I'll revert this.

Reply:
We previously discussed, that we wan to avoid those. Also we want logging as far up as possible. So maybe, since you can't always pass in one of our contexts with a logger, maybe you can just return the error and handle it in the caller?

Reply:
Fixed in https://github.com/cloudfoundry-incubator/cf-operator/pull/642/commits/bf10a4553e1f9941c8767493434a0b2653fb4fa6.

Indeed it made way more sense to propagate the error. Any failure loading the DNS addon aliases is critical from the deployment perspective and should stop the whole thing.